### PR TITLE
[Java] Respect User Supplied Header Params

### PR DESCRIPTION
**Pull request by ckoegel**
Work done in #20032 for the Java `okhttp-gson` generator only.

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh ./bin/configs/*.yaml
  ./bin/utils/export_docs_generators.sh
  ``` 
  (For Windows users, please run the script in [Git BASH](https://gitforwindows.org/))
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  IMPORTANT: Do **NOT** purge/delete any folders/files (e.g. tests) when regenerating the samples as manually written tests may be removed.
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (upcoming `7.x.0` minor release - breaking changes with fallbacks), `8.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

**Comment by macjohnny** [comment]
cc Java - @bbdouglas (2017/07) @sreeshas (2017/08) @jfiala (2017/08) @lukoyanov (2017/09) @cbornet (2017/09) @jeff9finger (2018/01) @karismann (2019/03) @Zomzog (2019/04) @lwlee2608 (2019/10) @martin-mfg (2023/08)

File: samples/client/petstore/java/okhttp-gson-dynamicOperations/src/main/java/org/openapitools/client/api/PetApi.java [hunk 1]
```diff
@@ -272,6 +272,11 @@ public okhttp3.Call deletePetCall(Long petId, String apiKey, final ApiCallback _
             localVarHeaderParams.put("Content-Type", localVarContentType);
         }
 
+        if (apiKey != null) {
+            localVarHeaderParams.put("api_key", localVarApiClient.parameterToString(apiKey));
+        }
```
Comment (martin-mfg) [inline samples/client/petstore/java/okhttp-gson-dynamicOperations/src/main/java/org/openapitools/client/api/PetApi.java:277]: I think this change was not intended and should be prevented, right? I.e. when `dynamicOperations` is set to `true`, these headers should not be added.

**Review by martin-mfg — CHANGES_REQUESTED** [review]

File: samples/client/petstore/java/okhttp-gson-dynamicOperations/src/main/java/org/openapitools/client/api/PetApi.java (hunk 1 above)
Comment (ckoegel) [reply samples/client/petstore/java/okhttp-gson-dynamicOperations/src/main/java/org/openapitools/client/api/PetApi.java:277]: This was unintended, I'll remove them if `dynamicOperations` is true

**Comment by ckoegel** [comment]
circleci node0 is currently failing with the error: `Internal error in Develocity Maven extension: Failed to capture test events`
Do these just need to be re-run or could this be caused by something in my PR?

**Comment by macjohnny** [comment]
you can try merging to most recent master

**Comment by macjohnny** [comment]
@ckoegel can you merge the most recent master? https://github.com/OpenAPITools/openapi-generator/pull/20062 fixes the typescript test that fails

**Pull request merged**